ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/commons/parameters/TreeParameters.java

```diff
@@ -72,14 +50,32 @@ public double getTargetObjectiveValue() {
         return targetObjectiveValue;
     }
 
+    public UnoptimizedCnecParameters getUnoptimizedCnecParameters() {
+        return null;
+    }
+
+    public int getMaximumSearchDepth() {
+        return maximumSearchDepth;
+    }
+
+    public int getLeavesInParallel() {
+        return leavesInParallel;
+    }
+
     public static TreeParameters buildForPreventivePerimeter(@Nullable SearchTreeRaoParameters searchTreeRaoParameters) {
-        SearchTreeRaoParameters parameters = Objects.isNull(searchTreeRaoParameters) ? new SearchTreeRaoParameters() : searchTreeRaoParameters;
 
+        SearchTreeRaoParameters parameters = Objects.isNull(searchTreeRaoParameters) ? new SearchTreeRaoParameters() : searchTreeRaoParameters;
         switch (parameters.getPreventiveRaoStopCriterion()) {
             case MIN_OBJECTIVE:
-                return new TreeParameters(parameters, StopCriterion.MIN_OBJECTIVE, 0.0, parameters.getPreventiveLeavesInParallel());
+                return new TreeParameters(StopCriterion.MIN_OBJECTIVE,
+                    0.0, //unsused
```

Comment:
"unused" lets to think that the parameter is not used in the constructor.
replace "unused" by "value won't matter" or something